### PR TITLE
feat: multi-schema middleware and setup

### PR DESCRIPTION
**Pull request by rohit-clootrack**
Changes
* Add Tenant app to maintain schemas 
* Add custom middleware to extract X-Tenant-Id
* Add method called `set_custom_schema` to dynamically set the schema
* Add endpoint `/create_tenant` to create a new schema and run migrations
* Broke: tests might not work.
* Add [extension detail](https://chrome.google.com/webstore/detail/modheader-modify-http-hea/idgpnmonknjnojddfkpgkljpfnnfcklj)

### CODE REVIEW

File: {{cookiecutter.project_name}}/README.md [hunk 1]
```diff
@@ -152,17 +165,30 @@ docker exec -it neo_{{cookiecutter.project_slug}} bash
...
+```
+python3 manage.py create_tenant
+```
+
+Answer the prompted questions. 
+`schema_name` value is one you will have to pass `X-Tenant-Id` in the header for all the requests.
+
+6. Create a superuser
+```
+python3 manage.py createsuperuser
```
Comment (silverwrath) [inline {{cookiecutter.project_name}}/README.md]: isn't this `create_tenant_superuser`?

**Review by silverwrath — APPROVED** [review]
LGTM

File: {{cookiecutter.project_name}}/{{cookiecutter.project_slug}}/urls.py [hunk 2]
```diff
@@ -29,6 +30,7 @@
     path("", include(router.urls)),
     path("api-auth/", include("rest_framework.urls", namespace="rest_framework")),
     path("admin/", admin.site.urls),
+    path("create_tenant/", TenantListCreateAPIView.as_view(), name="create-tenant"),
```
Comment (rohit-clootrack) [inline {{cookiecutter.project_name}}/{{cookiecutter.project_slug}}/urls.py]: api/v1.0

File: {{cookiecutter.project_name}}/{{cookiecutter.project_slug}}/app/utils/middlewares.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        return hostname
+
+    def process_request(self, request):
+        # Connection needs first to be at the public schema, as this is where
+        # the tenant metadata is stored.
+
+        hostname = self.hostname_from_request(request)
+        if not hostname:
+            from django.http import JsonResponse
+            return JsonResponse({"error": "X-Tenant-Id header is missing"}, status=401)
```
Comment (rohit-clootrack) [inline {{cookiecutter.project_name}}/{{cookiecutter.project_slug}}/app/utils/middlewares.py]: check how to handle 500 if tenant is not correct

File: {{cookiecutter.project_name}}/{{cookiecutter.project_slug}}/app/utils/middlewares.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    connection.set_tenant(request.tenant)
+
+
+class TenantCustomMiddleware(TenantMainMiddleware):
+    @staticmethod
+    def hostname_from_request(request):
+        """Extracts hostname from request. Used for custom requests filtering.
+        By default removes the request's port and common prefixes.
+        """
+        hostname = request.headers.get("X-Tenant-Id", None)
```
Comment (rohit-clootrack) [inline {{cookiecutter.project_name}}/{{cookiecutter.project_slug}}/app/utils/middlewares.py]: make it all caps

**Pull request merged**